packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -176,7 +176,8 @@ export function entityDeserializer(
         ([_, field]) =>
           (field instanceof EdmTypeField ||
             field instanceof ComplexTypeField) &&
-          typeof json[field._fieldName] !== 'undefined'
+          typeof json[field._fieldName] !== 'undefined' &&
```

Comment:
perhaps you can create a little helper function: isUndefinedOrNull. Perhaps this exists already? Perhaps there is a build in funciton for this check?

Reply:
have you seen this @jjtang1985

Reply:
We treat undefined and null differently. So the helper function is not useful in this case.